### ISSUE TITLE
.should() is not working with cy.xpath()

### ISSUE THREAD

**Issue by yogitthakral** [cypress-io/cypress-xpath#29]
Getting Element Text for asserting or checking visibility of element is not working with cy.xpath()

Website I am trying cypress with : -
http://automationpractice.com/index.php (Just login and inspect the My Account Page heading as specified in below error too

Xpath Used :- //h1[@class='page-heading']

Error : -
CypressError: Timed out retrying: You attempted to make a chai-jQuery assertion on an object that is neither a DOM object or a jQuery object.

The chai-jQuery assertion you used was:

  > text

The invalid subject you asserted on was:

  > [<h1.page-heading>]

To use chai-jQuery assertions your subject must be valid.

This can sometimes happen if a previous assertion changed the subject.

**Comment by yogitthakral** [comment]
Same element if used with cy.get() is working fine , obviously we will have to define other selector for it e.g css- selector : h1.page-heading

**Comment by yogitthakral** [comment]
Dependency :- 

"cypress-xpath": "^1.3.0",

**Comment by bahmutov** [comment]
Can you show your test code?

**Comment by yogitthakral** [comment]
var genericFunctions = require('../common/genericFunctions.js')


class MyAccountPage extends genericFunctions {

    constructor() {
        super();
        this.myAccount_pageHeading = "h1.page-heading";
        this.myAccount_name = ".account > span";
        this.myAccount_welcome_message = "p.info-account";
        this.logout_Link = "a.logout";
    }

    click_logout_Link(){
        super.clickOnElement(logout_Link);
    }

    Assert_myAccount_pageHeading(message) {
        super.getWebElementFromCss(this.myAccount_pageHeading).should('have.text', message);
    }

    Assert_myAccount_name(message) {
         super.getWebElementFromCss(this.myAccount_name).should('have.text', message);
    }


    Assert_myAccount_welcome_message(message) {
         super.getWebElementFromCss(this.myAccount_welcome_message).should('contain', message);
    }

    isVisible_logout_Link() {
     super.getWebElementFromCss(this.logout_Link).should('be.visible');
    }




}
module.exports =  MyAccountPage;



here is this code , super class is the one using cy.xpath methods for clicking and all , and alternatively i have added getElementFromCss to make things work as a workaround

**Comment by yogitthakral** [comment]
here is the generic file : - 


class genericFunctions {



openUrl(url){
cy.visit(url);
}

getUrl(){
    return cy.url();
}

clickOnElement(xpath){
    cy.xpath(xpath).click();
}

type(element,valueToType){
cy.xpath(element).type(valueToType);
}

selectFromDown(xpath,valueToSelect){
    cy.xpath(xpath).select(valueToSelect)
}

getWebElementFromXpath(xpath){
    return cy.xpath(xpath)
}

getWebElementFromCss(cssSelector){
    return cy.get(cssSelector)
}

returnRandomEmail(){
    return `kiosk-${new Date().getTime()}@yopmail.com`
}






}

 module.exports = genericFunctions;

**Comment by bahmutov** [comment]
I still do not understand what the failing test is. Can you unwrap all the layers of extra code and create minimal reproduction, like

```js
cy.visit('url')
cy.xpath('//some xpath thing').should('be.visible')
```
Otherwise I have no idea what is failing in your tests

**Comment by yogitthakral** [comment]
Apologies for late reply , Yes , simple reproduction would be :- 

cy.visit('http://automationpractice.com/index.php')

//here goes the login function that makes user logged in to the website.After that :- 

cy.get('h1.page-heading').should('be.visible')
//is working fine while 

cy.xpath('//h1[@Class='page-heading']').should('be.visible') //is throwing the original error mentioned above.

is not working for me.

**Comment by bahmutov** [comment]
it might be fixed by https://github.com/cypress-io/cypress-xpath/pull/35 that has landed and published as v1.3.1

**Comment by bahmutov** [comment]
if you can try with v1.3.1 or v1.4.0 I would appreciate it. For now closing the issue
